pkg/pgmodel/ingestor/trace_writer.go

```diff
@@ -16,8 +16,31 @@ import (
 const (
 	insertSchemaURLSQL = `INSERT INTO %s.schema_url (url) 
 		VALUES ($1) RETURNING (id)`
-	insertSpanNameSQL = `INSERT INTO %s.span_name (name) 
-		VALUES ($1) ON CONFLICT (name) DO UPDATE SET name=EXCLUDED.name RETURNING (id)`
+	insertServiceNameSQL = `INSERT INTO %s.tag (tag_type, key_id, key, value)
+		VALUES
+		(
+			%s.resource_tag_type(),
+			1,
+			'service.name',
+			to_jsonb($1::text)
+		)
+		ON CONFLICT (key, value) DO NOTHING`
+	insertSpanNameSQL = `INSERT INTO %s.span_name (service_name_id, span_name, span_kind) 
+		SELECT
+			t.id,
+			$1,
+			$2
+		FROM %s.tag t
+		WHERE t.key = 'service.name'
+		AND t.value = to_jsonb($3::text)
+		ON CONFLICT (service_name_id, span_name, span_kind) DO NOTHING;`
+	selectSpanNameIdSQL = `
+		SELECT n.id
+		FROM %s.span_name n
+		INNER JOIN %s.tag t on (n.service_name_id = t.id and t.key = 'service.name')
+		WHERE n.span_name = $1
+		AND n.span_kind = $2
+		AND t.value = to_jsonb($3::text)`
```

Comment:
can we encapsulate all this logic in a plpgsql function that just take a service_name, span_name, and type and returns an id?

Reply:
definitely. will do